shared/src/balancer/event_handler.rs

```diff
@@ -318,29 +378,64 @@ impl_event_retrieving! {
     pub BalancerV2WeightedPoolFactoryContract for balancer_v2_weighted_pool_factory
 }
 
+impl_event_retrieving! {
+    pub BalancerV2WeightedPool2TokensFactoryContract for balancer_v2_weighted_pool_2_tokens_factory
+}
+
+impl BalancerEventUpdater {
+    pub async fn get_latest_registry(&self) -> PoolRegistry {
+        let registry_1 = &self.weighted_pool_updater.lock().await.store;
+        let registry_2 = &self.two_token_pool_updater.lock().await.store;
+        registry_1.merge(registry_2)
+    }
+}
+
 #[async_trait::async_trait]
 impl Maintaining for BalancerEventUpdater {
     async fn run_maintenance(&self) -> Result<()> {
-        self.0.run_maintenance().await
+        self.two_token_pool_updater.run_maintenance().await?;
+        self.weighted_pool_updater.run_maintenance().await
```

Comment:
These could now be run together with `futures::future::join_all` although they are quite big tasks... going to try out both in #710

Reply:
> Results appear to show that it doesn't make much of a difference either way.

If we are behind a load balancer, these requests would be handled by different nodes and would speed things up. I think its worth joining these futures. You can use the `try_join` macro to do this quite easily:
```rust
futures::try_join!(
  self.two_token_pool_updater.run_maintenance(),
  self.weighted_pool_updater.run_maintenance(),
);
```